### PR TITLE
[SPARK-9574][Streaming]Remove unnecessary contents of spark-streaming-XXX-assembly jars

### PR DESCRIPTION
**Pull request by zsxwing**
Removed contents already included in Spark assembly jar from spark-streaming-XXX-assembly jars.

### CODE REVIEW

**Comment by srowen** [comment]
CC @harishreedharan

**Comment by tdas** [comment]
Jenkins, test this please.

**Comment by tdas** [comment]
@zsxwing Could you update this PR with MQTT fixes as well.

File: pom.xml [hunk 1]
```diff
@@ -432,7 +432,7 @@
         <version>${commons-lang3.version}</version>
       </dependency>
       <dependency>
-        <groupId>org.apache.commons</groupId>
```
Comment (tdas) [inline pom.xml]: Why this change?

Comment (tdas) [inline pom.xml]: This link says that common-lang was moved from common-lang group to org.apache.commons group. 
http://mvnrepository.com/artifact/commons-lang/commons-lang

Comment (zsxwing) [inline pom.xml]: This is incorrect. The correct groupId for commans-lang 2 is `commons-lang`. However, no place used this one previously so it didn't break anything.

Comment (zsxwing) [inline pom.xml]: > This link says that common-lang was moved from common-lang group to org.apache.commons group. 
> http://mvnrepository.com/artifact/commons-lang/commons-lang

Right. But it's under `commons-lang3`: http://mvnrepository.com/artifact/org.apache.commons/commons-lang3 since they cannot change the old artifact.

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please

**Comment by zsxwing** [comment]
retest this please.

**Comment by zsxwing** [comment]
retest this please

**Comment by tdas** [comment]
I am merging this to master and 1.5, thanks!

**Comment by shivaram** [comment]
@srowen @tdas It looks like this PR somehow changed the dependency tree and affected the documentation build that happens as a cron job on Jenkins. 

The error log at https://amplab.cs.berkeley.edu/jenkins/view/Spark-Packaging/job/Spark-Master-Docs/690/consoleFull is a run that happened after this PR and https://amplab.cs.berkeley.edu/jenkins/view/Spark-Packaging/job/Spark-Master-Docs/689/consoleFull is a run from before

It looks like this PR somehow introduces a dependency on netty 3.2.2 through mqtt, curator etc.
```
Note: Unresolved dependencies path:
   org.jboss.netty:netty:3.2.2.Final
   +-org.apache.zookeeper:zookeeper:3.4.5 ((com.typesafe.sbt.pom.MavenHelper) 
+- org.apache.curator:curator-client:2.4.0
+- org.apache.curator:curator-framework:2.4.0
+- org.apache.curator:curator-recipes:2.4.0 ((com.typesafe.sbt.pom.MavenHelper) 
+- org.apache.spark:spark-core_2.10:1.5.0-SNAPSHOT
+- org.apache.spark:spark-streaming_2.10:1.5.0-SNAPSHOT ((com.typesafe.sbt.pom.MavenHelper) 
+- org.apache.spark:spark-streaming-mqtt_2.10:1.5.0-SNAPSHOT 
+- org.apache.spark:spark-streaming-mqtt-assembly_2.10:1.5.0-SNAPSHOT
```

Did some exclude rules change or something like that ?

cc @rxin

**Comment by tdas** [comment]
The issues is actually SBT failing to resolve the artifact in the m2 cache
even though the artifact is present. Basically the ivy2 cache of jboss
needs to blown away.

I am slowly fixing the issue on the jenkins workers. Since I am not able to
access the Jenkins workers directly (ssh access not working for me), I am
leveraging the Spark SBT master build to blow away cached jboss. And
whatever worker it is running, that worker is getting fixed. Here is what
fixes it.

rm -rf /home/jenkins/.ivy2/cache/org.jboss*
rm -rf /home/jenkins/.ivy2/cache/junit*

I have it added in the Spark-SBT-master -
https://amplab.cs.berkeley.edu/jenkins/job/Spark-Master-SBT/configure
If you can login into the workers, would you mind running these two lines
on each of them?

**Comment by tdas** [comment]
Workers that are already fixed are (along with a successfully compiled
build):

worker-02:
https://amplab.cs.berkeley.edu/jenkins/job/Spark-Master-SBT/3288/AMPLAB_JENKINS_BUILD_PROFILE=hadoop2.3,label=spark-test/console
worker-06:
https://amplab.cs.berkeley.edu/jenkins/job/Spark-Master-SBT/3288/AMPLAB_JENKINS_BUILD_PROFILE=hadoop1.0,label=spark-test/console
worker-08:
https://amplab.cs.berkeley.edu/jenkins/job/Spark-Master-SBT/3288/AMPLAB_JENKINS_BUILD_PROFILE=hadoop2.0,label=spark-test/console

**Comment by shivaram** [comment]
Yeah I have ssh access. Let me give the other ones a go

**Comment by shivaram** [comment]
Well I have ssh access but no permission to delete these files. I think your approach will clear this out or I'll ping Shane tomorrow
